rq/job.py

```diff
@@ -327,12 +327,17 @@ def get_status(self, refresh: bool = True) -> JobStatus:
         Args:
             refresh (bool, optional): Whether to refresh the Job. Defaults to True.
 
+        Raises:
+            InvalidJobOperation: If refreshing and nothing is returned from the `HGET` operation.
+
         Returns:
             status (JobStatus): The Job Status
         """
         if refresh:
             status = self.connection.hget(self.key, 'status')
-            self._status = as_text(status) if status else None
+            if not status:
+                raise InvalidJobOperation(f"Failed to retrieve status for job: {self.id}")
```

Comment:
Mind updating the test to test this code path so it can pass Codecov's test?

Reply:
sure thing!